### PR TITLE
feat(route): added hackyournews

### PR DESCRIPTION
**Pull request by ftiasch**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/hackyournews
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [v2 Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [v2 路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [x] Documentation / 文档说明
  - [x] EN / 英文文档
  - [ ] CN / 中文文档
- [ ] Full text / 全文获取
  - [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制 
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施? 
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

HackYourNews.com, according to its description, it is AI summaries of the top HN stories.

### CODE REVIEW

File: lib/v2/hackyournews/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    const baseUrl = 'https://hackyournews.com';
+    const { data: response } = await got(String(baseUrl));
+    const $ = cheerio.load(response);
+
+    const item = $('tr.story')
+        .map((_, story) => {
+            const title = $(story).find('a').text().slice(0, -7);
+            const nextRow = $(story).next();
+            const metas = nextRow.text().trimStart().split('|');
+            const upvotes = Number.parseInt(metas[0].split(' poinrts')[0].trim());
```
Comment (TonyRL) [inline lib/v2/hackyournews/index.js:16]:
```suggestion
            const upvotes = parseInt(metas[0].split(' points')[0].trim());
```

File: lib/v2/hackyournews/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+const { parseDate } = require('@/utils/parse-date');
+
+module.exports = async (ctx) => {
+    const baseUrl = 'https://hackyournews.com';
+    const { data: response } = await got(String(baseUrl));
+    const $ = cheerio.load(response);
+
+    const item = $('tr.story')
+        .map((_, story) => {
+            const title = $(story).find('a').text().slice(0, -7);
```
Comment (TonyRL) [inline lib/v2/hackyournews/index.js:13]:
```suggestion
            const title = $(story).find('a').first().text();
```

File: lib/v2/hackyournews/radar.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'hackyournews.com': {
+        _name: 'HackYourNews',
+        '.': [
+            {
+                title: 'AI summaries of the top HN stories',
+                docs: 'https://docs.rsshub.app/routes/programming#hack-your-news',
+                source: ['/'],
+                target: '/',
```
Comment (TonyRL) [inline lib/v2/hackyournews/radar.js:9]:
```suggestion
                target: '/hackyournews',
```

File: lib/v2/hackyournews/radar.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'hackyournews.com': {
+        _name: 'HackYourNews',
+        '.': [
+            {
+                title: 'AI summaries of the top HN stories',
```
Comment (TonyRL) [inline lib/v2/hackyournews/radar.js:6]:
```suggestion
                title: 'Index',
```
This follows the heading level 3 of the docs.

File: lib/v2/hackyournews/index.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            const author = metas[0].split('by')[1].trim();
+            const pubDate = parseDate(metas[1].trim());
+            let category = [];
+            // NOTE: If the summary is not already proceeded, we cannot get the category.
+            if (metas.length === 5) {
+                category = [metas[2].trim(), metas[3].trim()];
+            }
+            const a = nextRow.find('a');
+            const link = a.attr('href');
+            const comments = Number.parseInt(a.text());
```
Comment (TonyRL) [inline lib/v2/hackyournews/index.js:26]:
```suggestion
            const comments = parseInt(a.text());
```

File: lib/v2/hackyournews/index.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
+// Require necessary modules
+const got = require('@/utils/got'); // a customised got
+const cheerio = require('cheerio'); // an HTML parser with a jQuery-like API
+const { parseDate } = require('@/utils/parse-date');
+
+module.exports = async (ctx) => {
+    const baseUrl = 'https://hackyournews.com';
+    const { data: response } = await got(String(baseUrl));
```
Comment (TonyRL) [inline lib/v2/hackyournews/index.js:8]:
```suggestion
    const { data: response } = await got(baseUrl);
```
`baseUrl` is a string already, no need to cast it again.

**Comment by ftiasch** [comment]
@TonyRL May I ask if there is anything which blocks this PR from merging?

**Review by TonyRL — COMMENTED** [review]
> @TonyRL May I ask if there is anything which blocks this PR from merging?

https://github.com/DIYgod/RSSHub/pull/13317/files#r1327985041

Narrow down the selector to remove the word `Archive`

https://github.com/DIYgod/RSSHub/pull/13317#discussion_r1327985306

https://github.com/DIYgod/RSSHub/blob/bf2f258799e3ed01a36c5d3a7fda4a112567a6d0/website/docs/routes/programming.md?plain=1#L483

docs: https://docs.rsshub.app/joinus/new-radar#title

**Review by Rivercityscooterhire — APPROVED** [review]

**Pull request merged**